core/MyProtocol.cpp

```diff
@@ -111,14 +111,30 @@ bool protocolMQTT2MyMessage(MyMessage &message, char *topic, uint8_t *payload,
                             const unsigned int length)
 {
 	char *str, *p;
-	uint8_t index = 0;
+	int8_t index = -1;
 	message.setSender(GATEWAY_ADDRESS);
 	message.setLast(GATEWAY_ADDRESS);
 	message.setEcho(false);
-	for (str = strtok_r(topic + strlen(MY_MQTT_SUBSCRIBE_TOPIC_PREFIX) + 1, "/", &p);
+
+	// The subscription prefix can contain wildcards, but only the + wildcard.
```

Comment:
Why is only the single-level + wildcard supported?
If we add support for wildcards, users will expect to be able to use multi-level # wildcards too.

Reply:
The MQTT multi-level wildcard (#) is only allowed at the end of the topic. Because the prefix is always followed by other levels it cannot contain the # wildcard, although "my" code wouldn't complain about it.

The code only checks the number of / characters.
The syntactical check is left to the MQTT broker. That was already the case and I think that is right. I have browsed the MySensors code and checked all code where the PREFIX is used and none of needed a change.